miner/worker.go

```diff
@@ -775,11 +785,237 @@ func (w *worker) updateSnapshot() {
+			PoseidonCodeHash: w.current.state.GetPoseidonCodeHash(*to),
+			CodeSize:         w.current.state.GetCodeSize(*to),
+		}
+	}
+
+	// currently still need to add storageTrace into the traces, but only with the coinbase proof
+	storageTrace := &types.StorageTrace{Proofs: make(map[string][]hexutil.Bytes)}
+	if _, exist := storageTrace.Proofs[coinbase.String()]; !exist {
+		proof, err := w.current.state.GetProof(coinbase)
+		if err != nil {
+			log.Error("Proof for coinbase not available", "coinbase", coinbase, "error", err)
+			// but we still mark the proofs map with nil array
+		}
+		wrappedProof := make([]hexutil.Bytes, len(proof))
+		for i, bt := range proof {
+			wrappedProof[i] = bt
+		}
+		storageTrace.Proofs[coinbase.String()] = wrappedProof
+	}
+
+	txStorageTrace := &types.StorageTrace{
+		RootBefore:    w.current.state.GetRootHash(),
+		Proofs:        make(map[string][]hexutil.Bytes),
+		StorageProofs: make(map[string]map[string][]hexutil.Bytes),
+	}
+
 	receipt, err := core.ApplyTransaction(w.chainConfig, w.chain, &coinbase, w.current.gasPool, w.current.state, w.current.header, tx, &w.current.header.GasUsed, *w.chain.GetVMConfig())
 	if err != nil {
 		w.current.state.RevertToSnapshot(snap)
 		return nil, err
 	}
+
+	txStorageTrace.RootAfter = w.current.state.GetRootHash()
+
+	proofAccounts := tracer.UpdatedAccounts()
+	if w.chainConfig.Scroll.FeeVaultEnabled() {
+		proofAccounts[*w.chainConfig.Scroll.FeeVaultAddress] = struct{}{}
+	} else {
+		proofAccounts[coinbase] = struct{}{}
+	}
+	proofAccounts[rcfg.L1GasPriceOracleAddress] = struct{}{}
+	for addr := range proofAccounts {
+		addrStr := addr.String()
+		_, existed := txStorageTrace.Proofs[addrStr]
+		if existed {
+			continue
+		}
+		proof, err := w.current.state.GetProof(addr)
+		if err != nil {
+			log.Error("Proof not available", "address", addrStr, "error", err)
+			// but we still mark the proofs map with nil array
+		}
+		wrappedProof := make([]hexutil.Bytes, len(proof))
+		for i, bt := range proof {
+			wrappedProof[i] = bt
+		}
+		txStorageTrace.Proofs[addrStr] = wrappedProof
+	}
+
+	proofStorages := tracer.UpdatedStorages()
+	proofStorages[rcfg.L1GasPriceOracleAddress] = vm.Storage(
+		map[common.Hash]common.Hash{
+			rcfg.L1BaseFeeSlot: {},
+			rcfg.OverheadSlot:  {},
+			rcfg.ScalarSlot:    {},
+		})
+	for addr, keys := range proofStorages {
+		if _, existed := txStorageTrace.StorageProofs[addr.String()]; !existed {
+			txStorageTrace.StorageProofs[addr.String()] = make(map[string][]hexutil.Bytes)
+		}
+
+		trie, err := w.current.state.GetStorageTrieForProof(addr)
+		if err != nil {
+			// but we still continue to next address
+			log.Error("Storage trie not available", "error", err, "address", addr)
+			continue
+		}
+		zktrieTracer := w.current.state.NewProofTracer(trie)
+
+		for key, values := range keys {
+			addrStr := addr.String()
+			keyStr := key.String()
+			isDelete := bytes.Equal(values.Bytes(), common.Hash{}.Bytes())
```

Comment:
here, i scan the codebase,  temporary  variable  `values` is `[HashLength]byte`. So check the values is empty, can just the array is empty? like 
```go
isDelete := len(values) == 0
```

Reply:
`common.Hash{}` is default empty hash.
I think it's easier to understand and more readable

Reply:
ok